pkg/rtc/transport.go

```diff
@@ -2154,3 +2222,45 @@ func nonSimulcastRTXRepairsFromSDP(s *sdp.SessionDescription, logger logger.Logg
 
 	return rtxRepairFlows
 }
+
+// ----------------------
+
+type iceCandidatePairStatsEncoder struct {
+	stats []iceCandidatePairStats
+}
+
+func (e iceCandidatePairStatsEncoder) MarshalLogArray(arr zapcore.ArrayEncoder) error {
+	for _, s := range e.stats {
+		if err := arr.AppendObject(s); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type iceCandidatePairStats struct {
+	webrtc.ICECandidatePairStats
+	local, remote webrtc.ICECandidateStats
+}
+
+func (r iceCandidatePairStats) MarshalLogObject(e zapcore.ObjectEncoder) error {
+	candidateToString := func(c webrtc.ICECandidateStats) string {
+		return fmt.Sprintf("%s:%d %s type(%s/%s), priority(%d)", c.IP, c.Port, c.Protocol, c.CandidateType, c.RelayProtocol, c.Priority)
+	}
+	e.AddString("state", string(r.State))
+	e.AddBool("nominated", r.Nominated)
+	e.AddString("local", candidateToString(r.local))
+	e.AddString("remote", candidateToString(r.remote))
```

Comment:
We have been logging public IPs with truncation for PII. Pion does not do that. So, if we have Pion logs with IP, it will do the full thing. But, wondering if we should truncate here. That does reduce the usefulness of these logs though.